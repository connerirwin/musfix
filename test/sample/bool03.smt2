; Sorts
(declare-sort Str 0)

; Functions
(declare-const cat Str)
(declare-const not_the_hippopotamus Str)
(declare-const dog Str)

; Qualifiers
; (qualif LE ((v @a0)(x @a0)) (<= x v))
; I have no idea how you are supposed to infer this, but casting bools to int
; and then performing <= is the same as checking implies
(qualif Implies ((v Bool) (x Bool)) (=> x v))

; Well-formedness constraints
(wf $k1 ((bx Bool)(v Bool)))

; Horn constraints
(constraint (forall ((bx Bool) (v1 Bool)) (=> (and True (=> bx v1)) ($k1 bx v1))))
(constraint (forall ((bx Bool) (zx Bool) (v2 Bool)) (=> (and True (and (not (= cat dog)) ($k1 bx v2))) (=> bx v2))))
