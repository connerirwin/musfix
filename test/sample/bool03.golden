
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/./sample/bool03.smt2
SortDecl "Str" 0
UninterpFunction "cat" [] (DataS "Str" [])
UninterpFunction "not_the_hippopotamus" [] (DataS "Str" [])
UninterpFunction "dog" [] (DataS "Str" [])
Qualifier "Implies" [Var BoolS "v",Var BoolS "x"] (Binary Implies (Var BoolS "x") (Var BoolS "v"))
WFConstraint "$k1" [Var BoolS "bx",Var BoolS "v"]
HornConstraint [Var BoolS "bx",Var BoolS "v1"] (Binary Implies (Binary And (BoolLit True) (Binary Implies (Var BoolS "bx") (Var BoolS "v1"))) (Unknown (fromList [("bx",Var BoolS "bx"),("v",Var BoolS "v1")]) "$k1"))
HornConstraint [Var BoolS "bx",Var BoolS "zx",Var BoolS "v2"] (Binary Implies (Binary And (BoolLit True) (Binary And (Unary Not (Binary Eq (Func (DataS "Str" []) "cat" []) (Func (DataS "Str" []) "dog" []))) (Unknown (fromList [("bx",Var BoolS "bx"),("v",Var BoolS "v2")]) "$k1"))) (Binary Implies (Var BoolS "bx") (Var BoolS "v2")))

QMAP
--------
fromList [("$k1",QSpace {_qualifiers = [Binary Implies (Var BoolS "v") (Var BoolS "bx"),Binary Implies (Var BoolS "bx") (Var BoolS "v")], _maxCount = 2})]

Candidates
--------
[Candidate {solution = fromList [("$k1",fromList [Binary Implies (Var BoolS "bx") (Var BoolS "v")])], validConstraints = fromList [Binary Implies (Binary And (BoolLit True) (Binary And (Unary Not (Binary Eq (Func (DataS "Str" []) "cat" []) (Func (DataS "Str" []) "dog" []))) (Unknown (fromList [("bx",Var BoolS "bx"),("v",Var BoolS "v2")]) "$k1"))) (Binary Implies (Var BoolS "bx") (Var BoolS "v2")),Binary Implies (Binary And (BoolLit True) (Binary Or (Unary Not (Var BoolS "bx")) (Var BoolS "v1"))) (Unknown (fromList [("bx",Var BoolS "bx"),("v",Var BoolS "v1")]) "$k1")], invalidConstraints = fromList [], label = "0"}]


Final candidates: 
0: [$k1 -> {bx [37m==>[0m v}] (2 0)

